Python/bytecodes.c

```diff
@@ -5027,6 +5027,35 @@ dummy_func(
             assert(tstate->tracing || eval_breaker == FT_ATOMIC_LOAD_UINTPTR_ACQUIRE(_PyFrame_GetCode(frame)->_co_instrumentation_version));
         }
 
+        inst(CHECK_ITERABLE, (iterable_st -- iterable_st)) {
+            PyObject *iterable = PyStackRef_AsPyObjectBorrow(iterable_st);
+            PyTypeObject *type = Py_TYPE(iterable);
+            if (oparg == 0) {
+                // Sync case, similar to GET_ITER
+                if (type->tp_iter == NULL) {
+                    if (!PySequence_Check(iterable)) {
+                        PyErr_Format(PyExc_TypeError,
+                            "'%.200s' object is not iterable",
+                            type->tp_name);
+                        ERROR_NO_POP();
+                    }
+                }
+            } else if (oparg == 1) {
+                // Async case, similar to GET_AITER
+                unaryfunc getter = NULL;
+                if (type->tp_as_async != NULL) {
+                    getter = type->tp_as_async->am_aiter;
+                }
+                if (getter == NULL) {
```

Comment:
Hmm, what's the point of the extra hoops with `getter` here? Ideally, we could just shorten this down to:

```suggestion
                if (type->tp_as_async == NULL || type->tp_as_async->am_aiter == NULL) {
```